### DISCUSSION TITLE
[feature request] Highlight for Image and CurrentLayoutIcon widget

### DISCUSSION THREAD

**Discussion by Crystal4276** [qtile/qtile#3994] [Ideas]
Hi,

I noticed that several qtile widgets have some highlight methods but not the image widget.
It would be nice to have those additional options.
Highlight methods would be the "usual" : `block`, `border`, `line` and one extra for that particular case `image`.

`Image` highlight was recently implemented in qtile extra for popupimage. This allows to switch to a different image file when the widget is highlighted.

EDIT: CurrentLayoutIcon could receive the same treatment

**Comment by elParaguayo** [comment]
So you just want a different image when the pointer is over the image widget?

I'm not sure this is something that would go in the main qtile repo as I'm not quite sure of the point of it. The `Image` widget just displays a static image. It has an `update` method which allows you to change the image.

I've actually got a PR for an updated version of the widget in qtile-extras as I've modified it to be able to render images from a URL (I want this to display artwork from media players) so I could add this highlight feature there too.

However, if @m-col likes the idea then it's pretty simple to add the highlight functionality to the main repo instead.

**Reply by Crystal4276** [reply]
> I'm not sure this is something that would go in the main qtile repo as I'm not quite sure of the point of it.

The main motivation is because i'm using the image widget coupled to mouse callback.
Highlighting the icon would indicate to the user that there is an action available.

For instance the Arch icon image is used here to open rofi.
![Screenshot from 2022-11-14 13-06-53](https://user-images.githubusercontent.com/27446196/201633208-0edd38d9-88e6-436e-96d3-7fb1414b5230.png)

>It has an update method which allows you to change the image.

Oh cool, i didn't know.

**Reply by elParaguayo** [reply]
I realised just after posting my message that the reason you might want this was pretty obvious and exactly as you described!
